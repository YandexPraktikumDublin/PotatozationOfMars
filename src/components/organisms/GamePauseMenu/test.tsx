import React from 'react'
import { shallow } from 'enzyme'
import toJSON from 'enzyme-to-json'
import { Provider } from 'react-redux'
import { Store } from 'redux'
import configureStore from 'redux-mock-store'

import { GamePauseMenu } from '.'

describe('<GamePauseMenu />', () => {
  const mockStore = configureStore([])

  let store: Store

  const initialState = {
    game: {}
  }

  beforeEach(() => {
    store = mockStore(initialState)
  })

  const toggleModal = jest.fn()

  it('should renders correct <GamePauseMenu />', () => {
    const wrapper = shallow(
      <Provider store={store}>
        <GamePauseMenu toggleModal={toggleModal} />
      </Provider>
    )

    expect(toJSON(wrapper)).toMatchSnapshot()
  })
})
