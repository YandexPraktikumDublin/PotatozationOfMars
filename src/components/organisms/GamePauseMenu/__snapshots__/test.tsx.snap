// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GamePauseMenu /> should renders correct <GamePauseMenu /> 1`] = `
<Fragment>
  <Button
    onClick={[MockFunction]}
  >
    Resume
  </Button>
  <Button
    onClick={[Function]}
  >
    Settings
  </Button>
  <Button
    onClick={[Function]}
  >
    Quit
  </Button>
</Fragment>
`;
